apps/openmw/mwmechanics/character.cpp

```diff
@@ -594,9 +594,9 @@ void getWeaponGroup(WeaponType weaptype, std::string &group)
 }
 
 
-MWWorld::ContainerStoreIterator getActiveWeapon(CreatureStats &stats, MWWorld::InventoryStore &inv, WeaponType *weaptype)
+MWWorld::ContainerStoreIterator getActiveWeapon(CreatureStats &stats, MWWorld::InventoryStore &inv, WeaponType *weaptype, bool getRealWeapon)
 {
-    if(stats.getDrawState() == DrawState_Spell)
+    if(!getRealWeapon && stats.getDrawState() == DrawState_Spell)
```

Comment:
The 'getRealWeapon' parameter seems a bit messy, is there any way to refactor the code so we don't need it?